pkg/reporter/github_reporter.go

```diff
@@ -204,6 +157,43 @@ func (g *GitHubReporter) CreateStatus(ctx context.Context, st Status, p *Params)
 	return nil
 }
 
+// ClearStatus deletes all github comments with the guardian prefix.
+func (g *GitHubReporter) ClearStatus(ctx context.Context) error {
+	listOpts := &gh.IssueListCommentsOptions{
+		ListOptions: gh.ListOptions{PerPage: 100},
+	}
+
+	for {
+		response, err := g.gitHubClient.ListIssueComments(ctx, g.inputs.GitHubOwner, g.inputs.GitHubRepo, g.inputs.GitHubPullRequestNumber, listOpts)
+		if err != nil {
+			return fmt.Errorf("failed to list comments: %w", err)
+		}
+
+		if response.Comments == nil {
+			return nil
+		}
+
+		for _, comment := range response.Comments {
+			// prefix is not found, skip
+			if !strings.HasPrefix(comment.Body, githubCommentPrefix) {
+				continue
+			}
+
+			// found the prefix, delete the comment
+			if err := g.gitHubClient.DeleteIssueComment(ctx, g.inputs.GitHubOwner, g.inputs.GitHubRepo, comment.ID); err != nil {
+				return fmt.Errorf("failed to delete comment: %w", err)
+			}
+		}
+
+		if response.Pagination == nil {
+			break
```

Comment:
Nit: there's an inconsistency here between `break` here at line 189 and `return nil` at line 173. I'd suggest picking one way of breaking from the loop and using it in both places.

Reply:
ah i migrated this and made it simpler, refactoring issue thanks!